### ISSUE TITLE
CommaDelimitedList Parameter is flagged 'is not of type 'array' when used in AWS SAM

### ISSUE THREAD

**Issue by alpotr** [aws-cloudformation/cfn-lint#4229]
### CloudFormation Lint Version

cfn-lint 1.39.0

### What operating system are you using?

Linux Fedora 34

### Describe the bug

Given the following template:

```
AWSTemplateFormatVersion: '2010-09-09'
Transform:
- AWS::LanguageExtensions
- AWS::Serverless-2016-10-31
Description: Test Serverless with LanguageExtensions
Parameters:
    VpcName:
        Type: String
        Default: MyVpc
    VPCCidr:
        Type: String
        Default: 10.0.0.0/16
    AZIds:
        Type: CommaDelimitedList
        Default: use1-az1,use1-az2

Resources:
  MyVpc:
    Type: AWS::EC2::VPC
    Properties:
      CidrBlock:
        Ref: VPCCidr
      EnableDnsSupport: 'true'
      EnableDnsHostnames: 'true'
      Tags:
      - Key: Name
        Value:
          Ref: VpcName

  MySubnet:
      Type: 'AWS::EC2::Subnet'
      Properties:
          VpcId: !Ref MyVpc
          CidrBlock: !Select
              - 1
              - !Cidr
                  - !Ref VPCCidr
                  - 6
                  - 8            
          AvailabilityZoneId: !Select [ 0, !Ref AZIds ] 
```

Parameter AZIds will be flagged with the below error, although _sam deploy_ will work

_E1017 'use1-az1,use1-az2' is not of type 'array'
template-test.yaml:15:18_

Removing the **AWS::Serverless-2016-10-31** from the Transform section or changing the code to the version below will resolve the error.

`!Select [ 0, !Split [",",!Ref AZIds ] ]`

### Expected behavior

My understanding is that when you use the _**!Ref**_ intrinsic function on a _**CommaDelimitedList**_ parameter, CloudFormation automatically converts the comma-separated string into an array of strings, so the need to use _!Split_ to create an array is not necessary.

### Reproduction template

```
AWSTemplateFormatVersion: '2010-09-09'
Transform:
- AWS::LanguageExtensions
- AWS::Serverless-2016-10-31
Description: Test Serverless with LanguageExtensions
Parameters:
    VpcName:
        Type: String
        Default: MyVpc
    VPCCidr:
        Type: String
        Default: 10.0.0.0/16
    AZIds:
        Type: CommaDelimitedList
        Default: use1-az1,use1-az2

Resources:
  MyVpc:
    Type: AWS::EC2::VPC
    Properties:
      CidrBlock:
        Ref: VPCCidr
      EnableDnsSupport: 'true'
      EnableDnsHostnames: 'true'
      Tags:
      - Key: Name
        Value:
          Ref: VpcName

  MySubnet:
      Type: 'AWS::EC2::Subnet'
      Properties:
          VpcId: !Ref MyVpc
          CidrBlock: !Select
              - 1
              - !Cidr
                  - !Ref VPCCidr
                  - 6
                  - 8            
          AvailabilityZoneId: !Select [ 0, !Ref AZIds ] 
```
